doc/rst/language/spec/string.rst

```diff
@@ -0,0 +1,180 @@
+
+An escaped data sequence can be reconstructed with ``~string.encode``:
+
+.. code-block:: chapel
+
+ var reconstructedBytes = myEscapedString.encode(policy=encodePolicy.unescape);
+ writeln(myBytes == reconstructedBytes);  // prints true
+
+Alternatively, escaped sequence can be used as-is without reconstructing the
+bytes:
+
+.. code-block:: chapel
+
+ var escapedBytes = myEscapedString.encode(policy=encodePolicy.pass);
+ writeln(myBytes == escapedBytes);  // prints false
+
+.. note::
+
+  Strings that contain escaped sequences cannot be directly used with
+  unformatted I/O functions such as ``writeln``. :ref:`Formatted I/O
+  <about-io-formatted-io>` can be used to print such strings with binary
+  formatters such as ``%|s``.
+
+.. note::
+
+  The standard :mod:`FileSystem`, :mod:`Path` and :mod:`IO` modules can use
+  escaped strings as described above for paths and file names.
+
+
+Lengths and Offsets in Unicode Strings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For Unicode strings, and in particular UTF-8 strings, there are several possible
+units for offsets or lengths:
+
+ * bytes
+ * codepoints
+ * graphemes
+
+Most methods on the Chapel string type currently work with codepoint units by
+default. For example, ``string.size`` returns the length in codepoints
+and `int` values passed into ``string.this`` are offsets in codepoint
+units.
+
+It is possible to indicate byte or codepoint units for indexing in the
+string methods by using arguments of type ``byteIndex`` or
+``codepointIndex`` respectively.
+
+For speed of indexing with their result values, ``string.find()``
+and ``string.rfind()``` return a ``byteIndex``.
+
+.. note::
+
+  Support for grapheme units is not implemented at this time.
+
+Using the ``byteIndex`` and ``codepointIndex`` types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A value of type ``byteIndex`` or ``codepointIndex`` can be passed to certain
+*string* functions to indicate that the function should operate with units of
+bytes or codepoints. Passing a ``codepointIndex`` has the same behavior as
+passing an integral type. See ``string.this`` for an example.
+
+Both of these types can be created from an ``int`` via assignment or cast. They
+also support addition and subtraction with ``int``. Finally, values of same
+types can be compared.
+
+For example, the following function returns a string containing only the second
+byte of the argument:
+
+   .. code-block:: chapel
+
+     proc getSecondByte(arg:string) {
+       var offsetInBytes = 1:byteIndex;
+       return arg[offsetInBytes];
+     }
+
+Whereas the following function returns a string containing only the second
+codepoint of the argument:
+
+   .. code-block:: chapel
+
+     proc getSecondCodepoint(arg:string) {
+       var offsetInCodepoints = 1:codepointIndex;
+       return arg[offsetInCodepoints];
+     }
+
+
+.. String_Methods:
+
+String Methods
+~~~~~~~~~~~~~~
+
+The *string* type:
+
+.. include:: /builtins/String.rst
```

Comment:
You didn't cause this but if you wanted to fix the way the first return line of `isUpper` displays and get the whitespace in the documentation for `isSpace` to display correctly, that'd be awesome

Reply:
It looks like `isUpper` is fixed now, thanks!  But `isSpace` still is only showing the letters for `\t` instead of the slash